src/pages/workflows/workflow/common/PermissionsModal.tsx

```diff
@@ -0,0 +1,225 @@
+          }}
+        />
+      </div>
+      <Clickable
+        tooltip='Remove'
+        onClick={() => {
+          const newPermissions = _.remove({ user }, userPermissions);
+          setUserPermissions(newPermissions);
+        }}
+      >
+        <Icon icon='times' size={20} style={{ marginRight: '0.5rem' }} />
+      </Clickable>
+    </div>
+  );
+};
+
+const CurrentUsers = (props: CurrentUserProps) => {
+  const list = useRef<HTMLDivElement>(null);
+  const { userPermissions } = props;
+  return (
+    <>
+      <div style={{ ...Style.elements.sectionHeader, margin: '1rem 0 0.5rem 0' }}>Current Users</div>
+      <div ref={list} role='list' style={styles}>
+        {_.flow(
+          _.remove(publicUser),
+          _.map((user) => <User key={`user ${user?.user}`} userEmail={user} {...props} />)
+        )(userPermissions)}
+      </div>
+    </>
+  );
+};
+
+export const PermissionsModal = (props: WorkflowPermissionsModalProps) => {
+  const { workflowOrNamespace, namespace, name, selectedSnapshot, setPermissionsModalOpen } = props;
+  const signal: AbortSignal = useCancellation();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [permissions, setPermissions] = useState<WorkflowsPermissions>([]);
+  const [working, setWorking] = useState(false);
+  const [loaded, setLoaded] = useState<boolean>(false);
+  const [originalPermissions, setOriginalPermissions] = useState<WorkflowsPermissions>([]);
+  const userEmails = _.map('user', permissions);
+
+  useOnMount(() => {
+    const loadWorkflowPermissions = async () => {
+      try {
+        const workflowPermissions: WorkflowsPermissions = await Ajax(signal)
+          .Methods.method(namespace, name, selectedSnapshot)
+          .permissions();
+        setPermissions(workflowPermissions);
+        setOriginalPermissions(workflowPermissions);
+        setLoaded(true);
+      } catch (error) {
+        await reportError('Error loading permissions.', error);
+        setPermissionsModalOpen(false);
+      }
+    };
+
+    loadWorkflowPermissions();
+  });
+
+  const addUser = (userEmail) => {
+    setSearchValue('');
+    setPermissions(append({ user: userEmail, role: 'READER' } as RawWorkflowsPermissions));
+  };
+
+  const save = withBusyState(setWorking, async () => {
+    const toBeDeleted = _.remove((entry) => userEmails.includes(entry.user), originalPermissions);
+
+    const permissionUpdates = [
+      ..._.flow(_.map(_.pick(['role', 'user'])))(permissions),
+      ..._.map(({ user }) => ({ user, role: 'NO ACCESS' }), toBeDeleted),
+    ];
+
+    try {
+      await Ajax(signal).Methods.method(namespace, name, selectedSnapshot).setPermissions(permissionUpdates);
+      setPermissionsModalOpen(false);
+    } catch (error) {
+      await reportError('Error loading permissions.', error);
+      setPermissionsModalOpen(false);
+    }
+  });
+
+  return (
+    <Modal
+      title={`Permissions for ${workflowOrNamespace} ${namespace}/${name}`}
```

Comment:
- I can definitely update the title to un-include the actual name of the workflow, but I since the `namespace/name` is at the end of the title, I could also potentially add ellipses to deal with any overflow. What do you think?
- I don't think that statement is true. From my testing, it seems like the permissions are on the entire workflow, and don't change when looking at a different snapshot. I'd be happy to mob if you're seeing a different behavior on your end!
- Yes, now that we've decided to use the word `method` as opposed to `workflow`, I'll update this naming as well.